webhooks/clusterclass_test.go

```diff
@@ -254,612 +213,406 @@ func TestClusterClassValidation(t *testing.T) {
+						Build()).
+				WithWorkerMachineDeploymentClass(
+					builder.MachineDeploymentClass("aa").
+						WithInfrastructureTemplate(
+							builder.InfrastructureMachineTemplate(metav1.NamespaceDefault, "infra1").Build()).
+						WithBootstrapTemplate(
+							refToUnstructured(refBadTemplate)).
+						Build()).
+				Build(),
 			old:       nil,
 			expectErr: true,
 		},
 		{
-			name: "create fail if bad template in machine deployment infrastructure",
-			in: &clusterv1.ClusterClass{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "default",
-				},
-				Spec: clusterv1.ClusterClassSpec{
-					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-					ControlPlane: clusterv1.ControlPlaneClass{
-						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
-					},
-					Workers: clusterv1.WorkersClass{
-						MachineDeployments: []clusterv1.MachineDeploymentClass{
-							{
-								Class: "aa",
-								Template: clusterv1.MachineDeploymentClassTemplate{
-									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
-									Infrastructure: clusterv1.LocalObjectTemplate{Ref: refBadTemplate},
-								},
-							},
-						},
-					},
-				},
-			},
+			name: "create fail if bad template in machineDeployment Infrastructure",
+			in: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithInfrastructureClusterTemplate(
+					builder.InfrastructureClusterTemplate(metav1.NamespaceDefault, "infra1").Build()).
+				WithControlPlaneTemplate(
+					builder.ControlPlaneTemplate(metav1.NamespaceDefault, "cp1").
+						Build()).
+				WithControlPlaneInfrastructureMachineTemplate(
+					builder.InfrastructureMachineTemplate(metav1.NamespaceDefault, "cpInfra1").
+						Build()).
+				WithWorkerMachineDeploymentClass(
+					builder.MachineDeploymentClass("aa").
+						WithInfrastructureTemplate(
+							refToUnstructured(refBadTemplate)).
+						WithBootstrapTemplate(
+							builder.BootstrapTemplate(metav1.NamespaceDefault, "bootstrap1").Build()).
+						Build()).
+				Build(),
 			old:       nil,
 			expectErr: true,
 		},
 
 		// bad apiVersion in ref tests
 		{
-			name: "create fail if bad apiVersion in control plane",
-			in: &clusterv1.ClusterClass{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "default",
-				},
-				Spec: clusterv1.ClusterClassSpec{
-					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-					ControlPlane: clusterv1.ControlPlaneClass{
-						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: refBadAPIVersion},
-					},
-					Workers: clusterv1.WorkersClass{
-						MachineDeployments: []clusterv1.MachineDeploymentClass{
-							{
-								Class: "aa",
-								Template: clusterv1.MachineDeploymentClassTemplate{
-									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
-									Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-								},
-							},
-						},
-					},
-				},
-			},
+			name: "create fail if bad template in InfrastructureCluster",
```

Comment:
Can we keep `bad apiVersion` in this case and below? (I think otherwise the name is equal to the test cases above)